modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/SqlAstTraverser.java

```diff
@@ -306,25 +306,32 @@ private void checkEqualityOperation(GridSqlOperation equalOp,
         Set<String> actLeftCols;
         Set<String> actRightCols;
 
+        boolean actPkLeft = pkLeft;
+        boolean actPkRight = pkRight;
+
         if (leftTbl.equals(leftTblAls))
             actLeftCols = leftCols;
-        else if (leftTbl.equals(rightTblAls))
+        else if (leftTbl.equals(rightTblAls)) {
             actLeftCols = rightCols;
+            actPkLeft = pkRight;
+        }
```

Comment:
What's about leftTbl and rightTbl names? 
You've swap table columns, but not the names.

Seems, you can just swap column names instead.

Reply:
@AMashenkov Thanks, nice catch. I've simplified this code even better after swap column names only.